src/app/content/highlights/components/cardUtils.ts

```diff
@@ -9,13 +9,32 @@ import { assertDefined } from '../../../utils/assertions';
 import { cardMarginBottom } from '../constants';
 import { HighlightData } from '../types';
 
-export const getHighlightOffset = (container: HTMLElement | undefined, highlight: Highlight) => {
+export const getHighlightOffset = (
+  container: HTMLElement | undefined,
+  highlight: Highlight,
+  expandableAncestorsSelector?: string
```

Comment:
I think this doesn't have to be a parameter and can be hardcoded in this function
also, you could use the `[data-type="solution"][aria-expanded="false"]` to narrow it down only to sections that are collapsed and then you don't have to calculate their getBoundingClientRect